src/controller.js

```diff
@@ -0,0 +1,148 @@
+            const shipPortIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+            const portElement = document.querySelector(`[data-port-index = '${shipPortIndex}']`);
+            const shipElement = document.querySelector('#ship');
+            shipElement.style.top = `${portElement.offsetTop + 28}px`
+            shipElement.style.left = `${portElement.offsetLeft - 40}px`
+        }
+        setSail() {
+            const ship = this.ship;
+            const currentPortIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+            const nextPortIndex = currentPortIndex +1;
+            const nextPortElement = document.querySelector(`[data-port-index='${nextPortIndex}']`);
+
+            if (!nextPortElement) {
+                this.renderMessage('You have reached your final destination.');
+                } else {
+            this.renderMessage(`Now departing ${ship.currentPort.name}`);
+            const shipElement = document.querySelector('#ship');
+            const sailInterval = setInterval(() => {
+                const shipLeft = parseInt(shipElement.style.left, 10);
+                if (shipLeft === (nextPortElement.offsetLeft - 40)) {
+                    ship.setSail();
+                    ship.dock();
+                    document.querySelector('#sailbutton').disabled = false;
+                                        
+                    this.renderMessage(`Docked in ${ship.currentPort.name}`);
+                    this.headsUp();
+
+                    clearInterval(sailInterval);
+                } 
+                shipElement.style.left = `${shipLeft + 1}px`;
+            },10);
+            }            
+        }
+        renderMessage(message) {
+            const messageElement = document.createElement('div');
+            messageElement.id = 'message';
+            messageElement.innerHTML = message;
+
+            const viewport = document.querySelector('#viewport');
+            viewport.appendChild(messageElement);
+
+            setTimeout(() => {
+                viewport.removeChild(messageElement);
+            }, 2000);
+        }
+        headsUp() {
+            const ship = this.ship;
+            const headsUpInfo = document.getElementById('headsUp');
+            const currentIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+            const nextIndex = currentIndex + 1;
+            if (!ship.currentPort) {
+                headsUpInfo.innerHTML = 'There are no ports in your itinerary'
+            } else if (!ship.itinerary.ports[nextIndex]) {
+                headsUpInfo.innerHTML = `Current Port: ${ship.currentPort.name} </br> This is the final stop on your voyage.`;
+            } else {
+                headsUpInfo.innerHTML = `Current Port: ${ship.currentPort.name} </br> Next Port: ${ship.itinerary.ports[nextIndex].name}`;
+            }
+        }
+        addPort() {
+            document.querySelector('#portForm').addEventListener('submit', (e) => {
```

Comment:
I would move creation of the event listener to your constructor, as you have done for `setSail`, then have this event listener call `addPort` i.e.
```
constructor(ship) {
  ...
  document.querySelector('#portForm').addEventListener('submit', (e) => {
    e.preventDefault();
    this.addPort();
  });
}
...
addPort() {
  ...
}
```